src/Elevator.java

```diff
@@ -15,9 +15,22 @@ public void run() {
 			RequestData requestData = scheduler.processRequest();
 			
 			int diff = requestData.getCurrentFloor() - requestData.getRequestedFloor();
+			if(diff<0) {
+				for(int i = requestData.getCurrentFloor(); i <requestData.getRequestedFloor(); i++) {
+					requestData.setCurrentFloor(i+1);

```

Comment:
I don't think its a good idea to set the current floor in the `requestData`. What I meant in my initial comment was that every elevator should have a property called `currentFloor`. (This would come clutch in further iterations as to decide which elevator should handle the request). Frankly I don't think the RequestData should be altered at all, it should just be read. 
I like the rest of the logic, I think we should give Elevator that `currentFloor` property and replace every `requestData.setCurrentFloor` to `this.setCurrentFloor`. And `requestData.getCurrentFloor()` should be `this.getCurrentFloor()`.

Reply:
Ah now I see! Sorry I misunderstood previously